dshr/dshr_mod.F90

```diff
@@ -213,6 +214,14 @@ subroutine dshr_init(gcomp, compname, sdat, mpicom, my_task, inst_index, inst_su
        inst_index=1
     endif
 
+    call NUOPC_CompAttributeGet(gcomp, name="write_restart_at_endofrun", value=cvalue, rc=rc)
```

Comment:
Why, `isPresent .and. isSet` not retried by the call. This could be the source of the issue that we faces under UFS. I think the correct version of this call need to be

```
call NUOPC_CompAttributeGet(gcomp, name="write_restart_at_endofrun", value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
```